backend/src/upload/file.processor.ts

```diff
@@ -0,0 +1,181 @@
+import { InjectQueue, Process, Processor } from '@nestjs/bull';
+import { Job, Queue } from 'bull';
+import * as process from 'process';
+import {
+  extract_measurements_data,
+  extractCoordinatesFromRSP,
+  find_surveys,
+} from './upload';
+import { UploadService } from './upload.service';
+import { extract_dashcam_image_data, extract_road_image_data } from './image';
+import * as path from 'path';
+import * as fs from 'fs';
+import * as unzip from 'unzip-stream';
+
+/**
+ * This class handles the file processing queue, managing the extraction
+ * processing of uploaded files and importing their data into a database.
+ * @author  Kerbourc'h, Liu
+ */
+@Processor('file-processing')
+export class FileProcessor {
+  constructor(
+    @InjectQueue('file-processing') private fileQueue: Queue,
+    private readonly service: UploadService,
+  ) {}
+
+  @Process('unzip-file')
+  async unzip(job: Job<{ filePath: string }>) {
+    try {
+      const { filePath } = job.data;
+      console.log(`Unzipping file: ${filePath}`);
+
+      // Create a temporary directory for unzipping
+      const tempUnzipPath = path.join(
+        './uploads',
+        `temp_${new Date().getTime()}`,
+      );
+      if (!fs.existsSync(tempUnzipPath)) {
+        fs.mkdirSync(tempUnzipPath, { recursive: true });
+      }
+
+      // Stream the zip file and extract to the temporary directory
+      fs.createReadStream(filePath)
+        .pipe(unzip.Extract({ path: tempUnzipPath }))
+        .on('error', (error) => {
+          console.error(`Error unzipping file: ${filePath}`, error);
+        })
+        .on('close', async () => {
+          await this.handleUnzippedContent(tempUnzipPath, filePath);
+        });
+    } catch (error) {
+      console.error(`Error processing file: ${job.data.filePath}`, error);
+    }
+  }
+
+  async handleUnzippedContent(tempUnzipPath, originalFilePath) {
+    // TODO: is it useful to move out of the temp dir? can't we just import the data even if the dir is in a temp dir?
```

Comment:
@liulw2022 do we need to move the folder (containing the survey) out of the temp dir? Can't we just import the data from the temp dir? (it is very slow to move big files)

Reply:
I'm using this temporary folder to solve the issue of having folders with the same name after unzipping. I tried other methods, but they all ended up merging folders with the same name. One approach I attempted was appending a timestamp if a folder with the same name was detected, but this led to nested folders like `originalname_timestamp/originalname`, which wasn't ideal. It seems the main issue is that the function has to unzip before it can check for folders with the same name, and I'm not sure if there's a better solution.

In practice, moving files within the same directory seems quite fast. Although, we might not even need to worry about that since the folders get deleted afterward.

Reply:
Sorry, I realized I misunderstood. The latest upload should have corrected it.